src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,105 @@
-import React from 'react';
+import {
+  ChangeEvent,
+  Dispatch,
+  FC,
+  FormEvent,
+  SetStateAction,
+  useState,
+} from 'react';
+import cn from 'classnames';
+
+import { addComment } from '../api/comments';
+
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+
+interface Props {
+  selectedPost: Post;
+  setComments: Dispatch<SetStateAction<Comment[]>>;
+}
+
+export const NewCommentForm: FC<Props> = ({ selectedPost, setComments }) => {
+  const [authorName, setAuthorName] = useState('');
+  const [authorNameError, setAuthorNameError] = useState(false);
+  const [authorEmail, setAuthorEmail] = useState('');
+  const [authorEmailError, setAuthorEmailError] = useState(false);
+  const [commentText, setCommentText] = useState('');
+  const [commentTextError, setCommentTextError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleInputChange = (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    switch (event.target.name) {
+      case 'name':
+        setAuthorNameError(false);
+        setAuthorName(event.target.value.trim());
+        break;
+
+      case 'email':
+        setAuthorEmailError(false);
+        setAuthorEmail(event.target.value.trim());
+        break;
+
+      case 'body':
+        setCommentTextError(false);
+        setCommentText(event.target.value.trim());
+        break;
+    }
+  };
+
+  const handleResetForm = () => {
+    setAuthorName('');
+    setAuthorEmailError(false);
```

Comment:
This line mistakenly sets `setAuthorEmailError(false)` twice. The first occurrence should be `setAuthorNameError(false)` to reset the name error state.